### DISCUSSION TITLE
Suddenly Error Previewing Query

### DISCUSSION THREAD

**Discussion by sjarifHD** [Budibase/budibase#4879] [Help] [answered]
Suddenly after update budibase builder to 1.0.82 (picture 1) every query I have with binding parameters becomes an error (picture 2), except for query without binding (picture 3)
![image](https://user-images.githubusercontent.com/9961361/157697000-1bdcbc02-422b-4974-9352-8989559c494b.png)
![image](https://user-images.githubusercontent.com/9961361/157696819-82f5048f-d1a7-4f49-a47d-70fcc589b30a.png)
![image](https://user-images.githubusercontent.com/9961361/157697441-db029b52-976a-459a-ad6b-77756e045276.png)

When I check docker logs for budibase-apps this is what I got
```
    at /app/dist/api/index.js:69:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
Trace: Error: error: bind message supplies 1 parameters, but prepared statement "" requires 0
    at PostgresIntegration.<anonymous> (/app/dist/integrations/postgres.js:138:27)
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/integrations/postgres.js:6:65)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  type: 'Error',
  '$error': '$error',
  expose: true,
  statusCode: 400,
  status: 400
}
    at /app/dist/api/index.js:69:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

```

**Comment by mike12345567** [comment]
Hi @sjarifHD - we recently updated how we perform SQL queries, could you possibly provide some more information about what database you're using and what version of it?

**Reply by sjarifHD** [reply]
I'm using PostgreSQL version 12.5

**Reply by sjarifHD** [reply]
For additional information, I have 3 budibase apps, 1 for playground and 2 for production used. I updated the playground app to the current release, and for production apps both of them not yet updated, but still got the same issues in those 3 apps

**Comment by sjarifHD** [comment, marked as answer]
For update, my team has been found solution for this problem. We need to change the query for pass binding to query. In my case `orderId` is string, before update it works wrapped in single-quote, after update 1.0.82 we need to remove single-quote.


Before:
```
select order_detail.*, order_master
from order_detail
inner join order_master on order_detail.order_id  = order_master.order_id
where order_master.order_id = '{{orderId}}'
````

After:
```
select order_detail.*, order_master
from order_detail
inner join order_master on order_detail.order_id  = order_master.order_id
where order_master.order_id = {{orderId}}
````

**Reply by mike12345567** [reply]
Hi @sjarifHD thanks for diagnosing this, fix in PR #4888.

**Reply by marco-santulli** [reply]
thank you for finding this @sjarifHD 
@mike12345567 sorry for bugging, but do we have a time for the fix to come online? because otherwise we have to go in and swap those ' in all our queries... or do we just take those out if they are not really needed from now on?

**Reply by marco-santulli** [reply]
one more bit, I took out the ' characters from one query to do some tests and get this error, although my query (an insert) worked

Trace: Datasource is not configured fully.
    at /app/dist/api/index.js:69:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
INFO  [1647183706314] (29 on 16f91c2fb2f1): request errored
    res: {
      "statusCode": 500,
      "headers": {
        "vary": "Accept-Encoding",
        "content-type": "application/json; charset=utf-8",
        "content-length": "14"
      }
    }
    err: {
      "type": "Error",
